### PR TITLE
Developer

### PR DESCRIPTION
**Pull request by RetroDavis**

### CODE REVIEW

File: NARFO_BE/NARFO_BE/Controllers/ActionsController.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            return await _context.Action.ToListAsync();
+        }
+
+        // GET: api/Actions/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Models.Action>> GetAction(int id)
+        {
+            var action = await _context.Action.FindAsync(id);
+
+            if (action == null)
```
Comment (RashadAkoodieRabbit) [inline NARFO_BE/NARFO_BE/Controllers/ActionsController.cs:36]: in c#, if you have one line for an if statement body, you have the option of doing away with the "{}"

File: NARFO_BE/NARFO_BE/Controllers/ActionsController.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                    throw;
+                }
+            }
+
+            return NoContent();
+        }
+
+        // POST: api/Actions
+        [HttpPost]
+        public async Task<ActionResult<Models.Action>> PostAction(Models.Action action)
```
Comment (RashadAkoodieRabbit) [inline NARFO_BE/NARFO_BE/Controllers/ActionsController.cs:72]: if you've already imported "Models", you don't have to keep calling the objects inside the namespace by *Models*.*

File: NARFO_BE/NARFO_BE/Controllers/ActivitiesController.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            }
+
+            return activity;
+        }
+
+        // PUT: api/Activities/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutActivity(int id, Activity activity)
+        {
+            if (id != activity.ActiviryId)
```
Comment (RashadAkoodieRabbit) [inline NARFO_BE/NARFO_BE/Controllers/ActivitiesController.cs:48]: spelling error. multiple instances. Refactor when changing it

File: NARFO_BE/NARFO_BE/Controllers/ActionsController.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!ActionExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
```
Comment (ash-rabbit) [inline NARFO_BE/NARFO_BE/Controllers/ActionsController.cs]: This should give you a 500 error so you will need to return something instead of throw...

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

File: NARFO_BE/NARFO_BE/Controllers/ActionsController.cs (hunk 4 above)
Comment (RetroDavis) [reply NARFO_BE/NARFO_BE/Controllers/ActionsController.cs]: show something like BadRequest( new { })  ?

Comment (Bongz12) [reply NARFO_BE/NARFO_BE/Controllers/ActionsController.cs]: Hie, is it bad when it throws 500

File: NARFO_BE/NARFO_BE/Controllers/ActionsController.cs (hunk 2 above)
Comment (soginga) [reply NARFO_BE/NARFO_BE/Controllers/ActionsController.cs:72]: Well it appears the System namespace contains an 'Action' class, so I was only resolving ambiguity between the two...

**Review by soginga — APPROVED** [review]

File: NARFO_BE/NARFO_BE/Controllers/ActivitiesController.cs (hunk 3 above)
Comment (soginga) [reply NARFO_BE/NARFO_BE/Controllers/ActivitiesController.cs:48]: someone fucked when designing the database..... we pulled that from the client's database tables

File: NARFO_BE/NARFO_BE/Controllers/ActionsController.cs (hunk 4 above)
Comment (Bongz12) [reply NARFO_BE/NARFO_BE/Controllers/ActionsController.cs]: we will change it to  return BadRequest(); for now,

**Review by CeeyarRbt — APPROVED** [review]

File: NARFO_BE/NARFO_BE/Controllers/ActionsController.cs (hunk 4 above)
Comment (ash-rabbit) [reply NARFO_BE/NARFO_BE/Controllers/ActionsController.cs]: I will re-open this issue until u fix it

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

**Pull request merged**